### PR TITLE
update sdk to match new kava 8 messages and queries

### PR DESCRIPTION
**Pull request by downing034**

### CODE REVIEW

File: example/swap.js [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+    await sleep(5000); // Wait 5 seconds
+
+    // claim swap rewards
+    const multiplierName = 'large';
+    const denomsToClaim = ['swp'];
+
+    const args = {"owner": Env.KavaAccount.Testnet.Address, "type": "swap"};
+    const claims = await kavaClient.getRewards(args);
+    if (claims) {
+        const claimRes = await kavaClient.claimSwapReward(
```
Comment (downing034) [inline example/swap.js:92]: this one doesn't live in the swap module so that's why it's not using `kavaClient.swap`

File: src/msg/swap/index.js [hunk 2]
```diff
@@ -0,0 +1,58 @@
+function newMsgDeposit(despositor, tokenA, tokenB, slippage, deadline) {
+  return {
+    type: 'swap/MsgDeposit'.
```
Comment (downing034) [inline src/msg/swap/index.js:3]: whoops! Have a `.` instead of a `,`. Will fix soon.

File: src/msg/swap/index.js [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    }
+  }
+}
+
+function newMsgSwapExactForTokens(requester, extactTokenA, tokenB, slippage, deadline) {
+  return {
+    type: 'swap/MsgSwapExactForTokens',
+    value: {
+      requester: requester,
+      extact_token_a: extactTokenA,
```
Comment (downing034) [inline src/msg/swap/index.js:32]: `extactTokenA ` should be `exactTokenA`

File: src/msg/swap/index.js [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+      token_b: tokenB,
+      slippage: slippate,
```
Comment (downing034) [inline src/msg/swap/index.js:34]: `slippate` should be `slippage`

File: src/msg/swap/index.js [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+  }
+}
+
+function newMsgSwapForExactTokens(requester, tokenA, exactTokenB, slippage, deadline) {
+  return {
+    type: 'swap/MsgSwapForExactTokens',
+    value: {
+      requester: requester,
+      token_a: tokenA,
+      extact_token_b: extactTokenB,
```
Comment (downing034) [inline src/msg/swap/index.js:46]: `extactTokenB ` should be `exactTokenB`

File: src/msg/swap/index.js [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+      slippage: slippate,
```
Comment (downing034) [inline src/msg/swap/index.js:47]: slippate should be slippage

File: src/msg/swap/index.js (hunk 3 above)
Comment (denalimarsh) [reply src/msg/swap/index.js:32]: `extact_token_a` -> `exact_token_a` as well

File: src/msg/swap/index.js (hunk 5 above)
Comment (denalimarsh) [reply src/msg/swap/index.js:46]: `extact_token_b` -> `exact_token_b` as well

**Review by denalimarsh — COMMENTED** [review]
Once you implement the changes from your comments this PR should be pretty much good to go. My revisions are pretty minor - updating examples, spelling in comments, etc.

**Comment by downing034** [comment]
@denalimarsh Let me know if you have any other feedback on this otherwise, I think we may be good to merge and publish to npm

File: src/utils/index.js [hunk 7]
```diff
@@ -157,10 +157,23 @@ const formatMultiCoins = (amounts, denoms) => {
   return coins;
 };
 
+/**
+ * Takes current day and adds desired amount of minutes and converts to unix time
+ * @param {number} minutes for the transaction to process
+ * @return {object} resulting in a unix timestring, defaulting to 1 day
+ */
+
+const calculateUnixTime = (minutes = 1440) => {
```
Comment (nddeluca) [inline src/utils/index.js:166]: Minutes is a bit strange imo - let's use seconds for the argument

File: example/swap.js [hunk 8]
```diff
@@ -0,0 +1,102 @@
...
+    kavaClient.setWallet(Env.KavaAccount.Testnet.Mnemonic);
+    await kavaClient.initChain();
+
+    // deposit coins to the swap bnb:usdx pool
+    const tokenA = kavaUtils.formatCoins(20, "bnb");
+    const tokenB = kavaUtils.formatCoins(10, "usdx");
+    const slippage = 0.01;
+    // create a deadline of 30 minutes from now for the max amount of time to wait
+    // for the transaction to be processed
+    const deadline = kavaUtils.calculateUnixTime(30);
```
Comment (nddeluca) [inline example/swap.js:17]: update to seconds

File: example/swap.js [hunk 9]
```diff
@@ -0,0 +1,102 @@
...
+var main = async () => {
+    // Start new Kava client
+    kavaClient = new KavaClient(Env.KavaEndpoints.Testnet);
+    kavaClient.setWallet(Env.KavaAccount.Testnet.Mnemonic);
+    await kavaClient.initChain();
+
+    // deposit coins to the swap bnb:usdx pool
+    const tokenA = kavaUtils.formatCoins(20, "bnb");
+    const tokenB = kavaUtils.formatCoins(10, "usdx");
+    const slippage = 0.01;
```
Comment (nddeluca) [inline example/swap.js:14]: Slippage must be a `string`

**Review by nddeluca — CHANGES_REQUESTED** [review]
Using seconds for the unix time helper is a bit more idiomatic - not common to pass minutes to functions, and I could see cases were a user would want to use a deadline under a minute (nice to use seconds as an integer and not a decimal then).

Slippage in the example is a `Number`, and never gets converted to a string when the message is created.  It looks like deadline is.

**Review by denalimarsh — CHANGES_REQUESTED** [review]
Changes LGTM! Noticed that we still need to update `newMsgVote` to include the `VoteType` as seen [here](https://github.com/Kava-Labs/kava/blob/master/x/committee/types/msg.go#L169).

**Review by denalimarsh — APPROVED** [review]

**Pull request merged**